File: aidp-python-client/src/aidp_python_client/aidataplatform_dp/models/repair_task_details.py
# coding: utf-8
# Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.



from oci.util import formatted_flat_dict, NONE_SENTINEL, value_allowed_none_or_none_sentinel  # noqa: F401
from oci.decorators import init_model_state_from_kwargs


@init_model_state_from_kwargs
class RepairTaskDetails(object):
    """
    Repair task details for a specific task selection.
    """

    def __init__(self, **kwargs):
        """
        Initializes a new RepairTaskDetails object with values from keyword arguments.
        The following keyword arguments are supported (corresponding to the getters/setters of this class):

        :param task_key:
            The value to assign to the task_key property of this RepairTaskDetails.
        :type task_key: str

        :param should_include_downstream:
            The value to assign to the should_include_downstream property of this RepairTaskDetails.
        :type should_include_downstream: bool

        :param should_include_upstream:
            The value to assign to the should_include_upstream property of this RepairTaskDetails.
        :type should_include_upstream: bool

        """
        self.swagger_types = {
            'task_key': 'str',
            'should_include_downstream': 'bool',
            'should_include_upstream': 'bool'
        }

        self.attribute_map = {
            'task_key': 'taskKey',
            'should_include_downstream': 'shouldIncludeDownstream',
            'should_include_upstream': 'shouldIncludeUpstream'
        }

        self._task_key = None
        self._should_include_downstream = None
        self._should_include_upstream = None

    @property
    def task_key(self):
        """
        **[Required]** Gets the task_key of this RepairTaskDetails.
        The display name of the task. User can specify a value for this.


        :return: The task_key of this RepairTaskDetails.
        :rtype: str
        """
        return self._task_key

    @task_key.setter
    def task_key(self, task_key):
        """
        Sets the task_key of this RepairTaskDetails.
        The display name of the task. User can specify a value for this.


        :param task_key: The task_key of this RepairTaskDetails.
        :type: str
        """
        self._task_key = task_key

    @property
    def should_include_downstream(self):
        """
        Gets the should_include_downstream of this RepairTaskDetails.
        Include transitive downstream tasks for this task selection.


        :return: The should_include_downstream of this RepairTaskDetails.
        :rtype: bool
        """
        return self._should_include_downstream

    @should_include_downstream.setter
    def should_include_downstream(self, should_include_downstream):
        """
        Sets the should_include_downstream of this RepairTaskDetails.
        Include transitive downstream tasks for this task selection.


        :param should_include_downstream: The should_include_downstream of this RepairTaskDetails.
        :type: bool
        """
        self._should_include_downstream = should_include_downstream

    @property
    def should_include_upstream(self):
        """
        Gets the should_include_upstream of this RepairTaskDetails.
        Include transitive upstream tasks for this task selection.


        :return: The should_include_upstream of this RepairTaskDetails.
        :rtype: bool
        """
        return self._should_include_upstream

    @should_include_upstream.setter
    def should_include_upstream(self, should_include_upstream):
        """
        Sets the should_include_upstream of this RepairTaskDetails.
        Include transitive upstream tasks for this task selection.


        :param should_include_upstream: The should_include_upstream of this RepairTaskDetails.
        :type: bool
        """
        self._should_include_upstream = should_include_upstream

    def __repr__(self):
        return formatted_flat_dict(self)

    def __eq__(self, other):
        if other is None:
            return False

        return self.__dict__ == other.__dict__

    def __ne__(self, other):
        return not self == other

File: aidp-java-client/src/main/java/com/oracle/aidataplatform/dp/model/RepairTaskDetails.java
// Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.

package com.oracle.aidataplatform.dp.model;



/**
 * Repair task details for a specific task selection.
**/
@jakarta.annotation.Generated(value = "OracleSDKGenerator", comments = "API Version: 20260430")
@com.fasterxml.jackson.databind.annotation.JsonDeserialize(builder=RepairTaskDetails.Builder.class)

public final class RepairTaskDetails  {
    @Deprecated
    @java.beans.ConstructorProperties({"taskKey", "shouldIncludeDownstream", "shouldIncludeUpstream"})
    public RepairTaskDetails(String taskKey, Boolean shouldIncludeDownstream, Boolean shouldIncludeUpstream) {
        super();
        this.taskKey = taskKey;
        this.shouldIncludeDownstream = shouldIncludeDownstream;
        this.shouldIncludeUpstream = shouldIncludeUpstream;
    }

    @com.fasterxml.jackson.databind.annotation.JsonPOJOBuilder(withPrefix = "")
    public static class Builder {
                /**
     * The display name of the task. User can specify a value for this.
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("taskKey")
private String taskKey;

        /**
         * The display name of the task. User can specify a value for this.
         * @param taskKey the value to set
         * @return this builder
         **/
        

public Builder taskKey(String taskKey) {
    this.taskKey = taskKey;
    return this;
}
            /**
     * Include transitive downstream tasks for this task selection.
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("shouldIncludeDownstream")
private Boolean shouldIncludeDownstream;

        /**
         * Include transitive downstream tasks for this task selection.
         * @param shouldIncludeDownstream the value to set
         * @return this builder
         **/
        

public Builder shouldIncludeDownstream(Boolean shouldIncludeDownstream) {
    this.shouldIncludeDownstream = shouldIncludeDownstream;
    return this;
}
            /**
     * Include transitive upstream tasks for this task selection.
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("shouldIncludeUpstream")
private Boolean shouldIncludeUpstream;

        /**
         * Include transitive upstream tasks for this task selection.
         * @param shouldIncludeUpstream the value to set
         * @return this builder
         **/
        

public Builder shouldIncludeUpstream(Boolean shouldIncludeUpstream) {
    this.shouldIncludeUpstream = shouldIncludeUpstream;
    return this;
}


        public RepairTaskDetails build() {
            RepairTaskDetails model = new RepairTaskDetails(this.taskKey
                , this.shouldIncludeDownstream
                , this.shouldIncludeUpstream);            return model;
        }

        @com.fasterxml.jackson.annotation.JsonIgnore
        public Builder copy(RepairTaskDetails model) {
                this.taskKey(model.getTaskKey());
    this.shouldIncludeDownstream(model.getShouldIncludeDownstream());
    this.shouldIncludeUpstream(model.getShouldIncludeUpstream());
return this;
        }
    }

    /**
     * Create a new builder.
     */
    public static Builder builder() {
        return new Builder();
    }


    public Builder toBuilder() {
        return new Builder().copy(this);
    }

    


        /**
     * The display name of the task. User can specify a value for this.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("taskKey")
    private final String taskKey;

        /**
     * The display name of the task. User can specify a value for this.
     * @return the value
     **/
    
    public String getTaskKey() {
        return taskKey;
    }


        /**
     * Include transitive downstream tasks for this task selection.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("shouldIncludeDownstream")
    private final Boolean shouldIncludeDownstream;

        /**
     * Include transitive downstream tasks for this task selection.
     * @return the value
     **/
    
    public Boolean getShouldIncludeDownstream() {
        return shouldIncludeDownstream;
    }


        /**
     * Include transitive upstream tasks for this task selection.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("shouldIncludeUpstream")
    private final Boolean shouldIncludeUpstream;

        /**
     * Include transitive upstream tasks for this task selection.
     * @return the value
     **/
    
    public Boolean getShouldIncludeUpstream() {
        return shouldIncludeUpstream;
    }

    @Override
    public String toString() {
        return this.toString(true);
    }

    /**
     * Return a string representation of the object.
     * @param includeByteArrayContents true to include the full contents of byte arrays
     * @return string representation
     */
    public String toString(boolean includeByteArrayContents) {
        java.lang.StringBuilder sb = new java.lang.StringBuilder();
        sb.append("RepairTaskDetails(");
        sb.append("taskKey=").append(String.valueOf(this.taskKey));
        sb.append(", shouldIncludeDownstream=").append(String.valueOf(this.shouldIncludeDownstream));
        sb.append(", shouldIncludeUpstream=").append(String.valueOf(this.shouldIncludeUpstream));
        sb.append(")");
        return sb.toString();
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) {
            return true;
        }
        if (!(o instanceof RepairTaskDetails)) {
            return false;
        }

        RepairTaskDetails other = (RepairTaskDetails) o;
        return java.util.Objects.equals(this.taskKey, other.taskKey) &&
            java.util.Objects.equals(this.shouldIncludeDownstream, other.shouldIncludeDownstream) &&
            java.util.Objects.equals(this.shouldIncludeUpstream, other.shouldIncludeUpstream);
    }

    @Override
    public int hashCode() {
        final int PRIME = 59;
        int result = 1;
        result = (result * PRIME) + (this.taskKey == null ? 43 : this.taskKey.hashCode());
        result = (result * PRIME) + (this.shouldIncludeDownstream == null ? 43 : this.shouldIncludeDownstream.hashCode());
        result = (result * PRIME) + (this.shouldIncludeUpstream == null ? 43 : this.shouldIncludeUpstream.hashCode());
        return result;
    }


}
